fathom/src/core/binary.rs

```diff
@@ -165,18 +165,16 @@ impl<'data> BufferReader<'data> {
     }
 
     /// Set the offset of the reader relative to the start of the backing buffer.
-    pub fn set_relative_offset(&mut self, relative_offset: usize) -> Result<(), ReadError> {
+    pub fn set_relative_offset(&mut self, relative_offset: usize) -> Option<()> {
         (relative_offset <= self.buffer.remaining_len())
             .then(|| self.relative_offset = relative_offset)
-            .ok_or(ReadError::SetOffsetOutsideBuffer)
     }
 
     /// Set the offset of the reader relative to the start position.
     pub fn set_offset(&mut self, offset: usize) -> Result<(), ReadError> {
-        match usize::checked_sub(offset, self.buffer.start_offset) {
-            Some(relative_offset) => self.set_relative_offset(relative_offset),
-            None => Err(ReadError::SetOffsetOutsideBuffer),
```

Comment:
Maybe could change to `ReadError::SetOffsetBeforeStartOfBuffer { offset }`. And `ReadError::SetOffsetAfterEndOfBuffer { offset: Option<T> }` (the `Option` to handle the exceedingly unlikely chance of an overflow). Then both `set_relative_offset` and `set_offset` can return results, with better errors.